Vera Simon - Intro to Python (day 1)

VeraSimon opened the pull request:
All the toys!

- [x] `hello` -- Hello world
- [x] `bignum` -- Print some big numbers
- [x] `datatypes` -- Experiment with type conversion
- [x] `modules` -- Learn to import from modules
- [x] `printing` -- Formatted print output
- [x] `lists` -- Python's version of arrays
- [x] `tuples` -- Immutable lists typically for heterogenous data
- [x] `slices` -- Accessing parts of lists
- [x] `comprehensions` -- List comprehensions
- [x] `dictionaries` -- Dictionaries
- [x] `functions` -- Functions
- [x] `args` -- Arguments and Keyword Arguments
- [x] `scopes` -- Global, Local, and Non-Local scope
- [x] `file_io` -- Read and write from files
- [x] `cal` -- Experiment with module imports and implement a text-based calendar
- [x] `classes` -- Classes and objects

VeraSimon committed:
Added missing newline at end of README.md

VeraSimon committed:
hello.py
Completed base Hello World script

VeraSimon committed:
bignum.py
printed a very large number

VeraSimon committed:
datatypes.py
typecasted some stuff

VeraSimon committed:
modules.py
For the glory of module imports!

VeraSimon committed:
printing.py
Printed some things. The printf % format is super fidgety.

VeraSimon committed:
lists.py
Manipulated some lists

VeraSimon committed:
tuples.py
Printed a tuple (and int)

VeraSimon committed:
slices.py
slicey, slicey

VeraSimon committed:
comprehensions.py
the python doc examples for this were needlessly complex, but I got it done

VeraSimon committed:
dictionaries.py
Manipulated a list of dictionaries

VeraSimon committed:
functions.py
Wrote a function!

VeraSimon committed:
functions.py
Bit of a tweak to the function after realizing the value passed in was already cast as an int

VeraSimon committed:
args.py
args and kwargs and quaks and squaks

BrianARuff reviewed (commented):
Great work so far! I love that you have provided so much error checking in a lot of your functions. See if you can make the `# Output the two middle elements in the array: [1, 7]` in slices.py more dynamic as an added challenge when you get around to it. You are doing promising work so far on this 👍 keep it up!!

VeraSimon committed:
scopes.py
global and nonlocal, oh my!

VeraSimon committed:
file_io.py
Im a little teapot...

VeraSimon committed:
cal.py
Shiny text calendar with argparse support

VeraSimon committed:
classes.py
Prints everything, but out of order. Investigating.

VeraSimon committed:
classes.py
A bit hacky with building the list and formating everything as shown, but it prints with the formatting and in the order expected.
